chapter12/dcox/ex12_01

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/perl
```

Comment:
I had to wait on `harness -iz` so had a quick bash at a hash as I hadn't thought to try it that way

```perl
#!/usr/bin/perl

use strict;
use warnings;

sub file_report {
    my $file_name = shift;

    my %file_attributes;
    $file_attributes{'existant'} = ( -e $file_name );
    $file_attributes{'readable'} = ( -r $file_name );
    $file_attributes{'writeable'} = ( -w $file_name );
    $file_attributes{'executable'} = ( -x $file_name );

    my $output = "$file_name is: ";
    foreach my $attribute ( sort keys %file_attributes ) {
      $output .= "non-" if ( ! $file_attributes{$attribute} );
      $output .= "$attribute ";
    }

    print "$output\n";
}

foreach my $file (@ARGV) {
    file_report($file);
}
```